docs/current_docs/developer-guide/go/742930-services.mdx

```diff
@@ -0,0 +1,260 @@
+---
+slug: /developer-guide/go/742930/services
+displayed_sidebar: "current"
+toc_max_heading_level: 2
+title: "Services"
+---
+
+# Services
+
+Dagger Functions support service containers, enabling users to spin up additional long-running services (as containers) and communicate with those services from Dagger Functions.
+
+This makes it possible to:
+- Instantiate and return services from a Dagger Function, and then:
+  - Use those services in other Dagger Functions (container-to-container networking)
+  - Use those services from the calling host (container-to-host networking)
+- Expose host services for use in a Dagger Function (host-to-container networking).
+
+Some common scenarios for using services with Dagger Functions are:
+
+- Running a database service for local storage or testing
+- Running end-to-end integration tests against a service
+- Running sidecar services
+
+## Service containers
+
+Services instantiated by a Dagger Function run in service containers, which have the following characteristics:
+
+- Each service container has a canonical, content-addressed hostname and an optional set of exposed ports
+- Service containers are started just-in-time, de-duplicated, and stopped when no longer needed
+- Service containers are health checked prior to running clients
+
+## Work with service hostnames and ports
+
+Each service container has a canonical, content-addressed hostname and an optional set of exposed ports. You can query a service container's canonical hostname with the `Service` type's `Hostname` function.
+
+Here is an example of a Dagger Function that creates a service and returns its hostname:
```

Comment:
Fully agree - this started as a cute way to show a technical building block for how c2c networking works, but at this point I think it does more harm than good and it would be better to open with a realistic example that the user can recognize and play with

Reply:
Done